WcaOnRails/app/webpacker/components/Auth/Signup/index.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useRef, useState } from 'react';
+import {
+  Button, Card, CardHeader, Container, Form, FormField, Message,
+} from 'semantic-ui-react';
+import ReCAPTCHA from 'react-google-recaptcha';
+import { GOOGLE_RECAPTCHA_SITE_KEY } from '../../../lib/wca-data.js.erb';
+import I18n from '../../../lib/i18n';
+
+export default function Signup() {
+  // initial is a minor hack to prevent fomantic CSS messing up the UI.
+  const [initial, setInitial] = useState(true);
+  const [formValues, setFormValues] = useState({
+    email: {
+      error: null,
+      id: 'email',
+      value: '',
+    },
+    password: {
+      error: null,
+      id: 'password',
+      value: '',
+    },
+    confirmPassword: {
+      error: null,
+      id: 'confirmPassword',
+      value: '',
+    },
+  });
+  const [captchaError, setCaptchaError] = useState(false);
+  const recaptchaRef = useRef();
+  const csrfToken = document.querySelector('meta[name=csrf-token]').content;
+
+  const setFormError = (id, error) => {
+    setFormValues({
+      ...formValues,
+      [id]: {
+        ...formValues[id],
+        error,
+      },
+    });
+  };
+
+  const validation = {
+    email: () => {
+      const emailRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
```

Comment:
Can we move constants like regexes into their own file?

Reply:
Ya that would be better. Done.